packages/core/src/plugins/dataTable.ts

```diff
@@ -0,0 +1,85 @@
+import cheerio from 'cheerio';
+import { FrontMatter, PluginContext } from './Plugin';
+import md from '../lib/markdown-it';
+
+const CSS_FILE_NAME = 'dataTableAssets/datatables.min.css';
+const CSS_ADDITIONAL = 'dataTableAssets/datatables-additional.css';
+const JS_FILE_NAME = 'dataTableAssets/datatables.min.js';
+
+const initScript = `
+  <script>
+    function getTableOptions(el) {
+      const options = {};
+      if ($(el).hasClass('sortable-table')) {
+        options.searching = false;
+        options.paging = false;
+        options.info = false;
+      } else if ($(el).hasClass('searchable-table')) {
+        options.ordering = false;
+        options.paging = false;
+        options.info = false;
+        options.dom = '<"row"<"col-sm-12"f>>' + '<"row"<"col-sm-12"t>>';
+      } else if ($(el).hasClass('sortable-searchable-table')) {
+        options.paging = false;
+        options.info = false;
+        options.dom = '<"row"<"col-sm-12"f>>' + '<"row"<"col-sm-12"t>>';
+      }
+      return options;
+    }
+
+    Vue.directive('datatable', {
+      inserted: function(el, binding) {
+        const options = binding.value || {};
+        const tableOptions = getTableOptions(el);
+        $(el).DataTable({ ...tableOptions, ...options });
+      }
+    });
+
+    document.addEventListener('DOMContentLoaded', function() {
+      $('table.sortable-table, table.searchable-table, table.sortable-searchable-table').each(function() {
+        const options = getTableOptions(this);
+        $(this).DataTable(options);
+      });
+    });
+  </script>
+`;
+
+export = {
+  getLinks: () => [
+    `<link rel="stylesheet" href="${CSS_FILE_NAME}">`,
+    `<link rel="stylesheet" href="${CSS_ADDITIONAL}">`,
+  ],
+  getScripts: () => [`<script src="${JS_FILE_NAME}"></script>`, initScript],
+  postRender: (pluginContext: PluginContext, frontmatter: FrontMatter, content: string) => {
+    const $ = cheerio.load(content);
+
+    $('d-table').each((index: number, node: cheerio.Element) => {
+      const $node = $(node);
+      const html = $node.html();
+      if (html != null) {
```

Comment:
My opinion is it would be cleaner to use a guard clause here. i.e. check for `html == null` instead of `html != null` and break/return. This allows the main logic of the code to not be wrapped in the if block, making it cleaner/more readable.

Here's a useful article on guard clauses: [https://deviq.com/design-patterns/guard-clause](https://deviq.com/design-patterns/guard-clause)